modules/ROOT/pages/apikit-odatav4-expand-tutorial.adoc

```diff
@@ -0,0 +1,162 @@
+
+* The `Employee` entity has a related `Manager` entity.
+* The `Employees` entity set is a collection of `Employee` entities.
+* The `Managers` entity set is a collection of `Manager` entities.
+* Our API is placed on `https://example.com/api/`
+
+Given this setup, the retrieval of employees with their managers requires:
+
+* `GET https://example.com/api/Employees` to retrieve a list of employees.
+* Gather the `Manager` ids through the `ReferentialConstraint` of their
+  relationship.
+* `GET https://example.com/api/Managers?$filter=ManagerID in (id1, id2, ..., idN)`
+  to retrieve a list of related managers.
+* Write a small program to merge the lists based on a join criteria.
+
+If you use `$expand` this can be shortened to:
+
+* `GET https://example.com/api/Employees?$expand=Manager` to retrieve a list of
+  employees with their `Manager` navigation property expanded.
+
+====
+
+`$expand` is a powerful system query option to implement. It reduces the amount
+of incoming network traffic and enables an easier to use API for consumers. The
+ability to request _"just what's needed"_ without multiple network roundtrips
+is useful for both metered consumers and latency-sensitive ones.
+
+== The _generic expand_ feature
+
+_Generic expand_ is the ability to reuse existent flow implementations as a
+starting point for expand requests. This allows a Mule Developer to focus only
+on the relevant data retrieval operations while the module performs the
+necessary steps to ensure `$expand`-ed fields get completed.
+
+From the point of view of a Mule Developer, an application with _generic expand_
+enabled is exactly like any other OData Mule Application. Flow execution
+differs: once the information for the main entity set is retrieved, the module
+will perform additional _virtual_ requests gathering the information for any
+`expand`-ed fields. If errors arise during this expansion operation, the final
+behaviour is configurable (throw the error to the consumer or ignore it
+resolving a `null` value).
+
+== Requirements
+
+- APIKit for OData Module (version 1.3.0)
+- An `<entity-collection-source/>` implementation for each expandable entity.
+- Support for the `in` filter on those `<entity-collection-source/>` flows.
+
+== Implementation guidelines
+
+In order to use our generic expansion implementation each source has to be
+annotated with the relevant types that might be expanded from it. Keep in mind
+that enabling `$expand` in a completely unrestricted way will allow API clients
+to write arbitrarily complex queries, this can cause performance problems.
+
+If you wish to implement `$expand` by yourself for some relationships leave
+those out of the list on the "expandable navigation properties" option. Manual
+implementations inside the flow will be needed for navigation properties that
+don't use _generic expansion_.
+
+=== Implementation checklist
+
+. Check that your CSDL and Mule application meets the expected requirements:
```

Comment:
Is the application name "CSDL and Mule application"? Or are we talking about two different things: "CSDL" and "Mule application"?

Reply:
CSDL file and Mule application meet...